packages/framework/src/types/validator.types.ts

```diff
@@ -0,0 +1,28 @@
+import { JSONSchema } from 'json-schema-to-ts';
+import { ValidateFunction as AjvValidateFunction } from 'ajv';
+import { ParseReturnType } from 'zod';
+import { Schema } from './schema.types';
+
+export type ValidateFunction<T = unknown> = AjvValidateFunction<T> | ((inputs: T) => ParseReturnType<T>);
+
+export type ValidationError = {
+  path: string;
+  message: string;
+};
```

Comment:
An opinionated error structure that all validators must transform into. There's still a bit of work to normlize the error messages (i.e. ensure that "Required" in JSONSchema-ajv equates to ZodSchema "must have x property"), but we can tackle this later.